thunder/core/transforms.py

```diff
@@ -3736,9 +3737,99 @@ def backward_fn(saved_for_backward, cotangents):
     _update_backward_with_new_saved_for_backward(backward_trace, only_used_bw_saved_for_backward)
     forward_trace.set_provenance(TraceProvenance("Augmented forward pass"))
     backward_trace.set_provenance(TraceProvenance("Backward pass"))
+
+    enable_saved_for_backward_recomputation: None | bool = get_compile_option(
+        "enable_saved_for_backward_recomputation", "Enable save for backward tensors recomputation."
+    )
+    if enable_saved_for_backward_recomputation:
+        forward_trace, backward_trace = recompute_saved_for_backward(forward_trace, backward_trace)
+
     return ForwardBackwardTraces(forward_trace, backward_trace)
 
 
+def recompute_saved_for_backward(fwd_trace: Trace, bwd_trace: Trace) -> tuple[Trace, Trace]:
+    """Generates the pair of traces with rematerializaion of the saved-for-backward tensors.
+    Args:
+        fwd_trace (Trace): forward trace where to get the saved for backward from.
+        bwd_trace (Trace): backward trace where to recompute the saved for backward to.
+
+    Returns:
+        tuple[Trace, Trace]: A tuple containing the new forward and backward traces.
+    """
+
+    start_time_ns = time.perf_counter_ns()
+
+    saved_for_bw = get_saved_for_backward_tensors(fwd_trace)
+    fwd_trace_args = {variableify(j) for j in fwd_trace.args}
+    old_saved_for_bwd = {variableify(j) for j in saved_for_bw}
+
+    all_rematerializable = old_saved_for_bwd - fwd_trace_args
+
+    # Here will come the selection logic for what to materialize.
+    from collections import Counter
+
+    tensor_counter = Counter({v: unvariableify(v).numel * unvariableify(v).dtype.bytes for v in all_rematerializable})
+    print(f"DEBUG: out of {len(saved_for_bw)} sfb, {len(all_rematerializable)} are recomputable")
+    print(f"DEBUG: can save up to {tensor_counter.total()*1e-6:.3f} MB")
+    top_one_third = len(saved_for_bw) // 3
+    print(f"DEBUG: remat {top_one_third} tensors (top one third)")
```

Comment:
If you'd like to keep these messages they should be printed through a debug logger.
```suggestion
    top_one_third = len(saved_for_bw) // 3
```